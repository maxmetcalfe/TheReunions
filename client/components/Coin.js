import React, { Component } from "react";
import PropTypes from "prop-types";
import resetMap from "../utils/resetMap"
import constants from "../constants";

var CLASS_NAME = "coin";
var SELECTED_CLASS_NAME = "coin selected";

class Coin extends Component {

  constructor(props) {
    super(props);
    this.reunions = props.reunions;
  }

  componentDidMount() {
    this.map = document.getElementById("map");
  }

  selectMemberReunionsByType(e) {
    // Unselect all reunions on the map.
    resetMap(this.map);
    // Send the selected element to the parent App.
    this.props.setSelection(this);
    // Tag this coin as selected.
    this.setState({selected: true});
    var self = this;
    e.reunions.forEach(function(reunion) {
      if (constants.CATEGORIES[reunion.properties.category] === self.props.type) {
        var marker = document.getElementById(reunion.properties.element_id);
        marker.classList.add("picked");
      }
    })
  }

  className() {
    if (this.displayAsSelected()) {
      return SELECTED_CLASS_NAME + " " + this.props.type;
    }
    return CLASS_NAME + " " + this.props.type;
  }

  displayAsSelected() {
    if (this.props.selection && this.props.id === this.props.selection.props.id) {
      return true;
    }
    return false;
  }

  render() {
    return (
      <div className={this.className()} onClick={()=>{this.selectMemberReunionsByType(this.props)}}>{this.props.counts}</div>
    );
  }
}

Coin.propTypes = {
  counts: PropTypes.number,
  reunions: PropTypes.array,
  selection: PropTypes.object,
  setSelection: PropTypes.func,
  type: PropTypes.string.isRequired,
  id: PropTypes.string
};

export default Coin;
