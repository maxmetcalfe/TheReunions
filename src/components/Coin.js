import React, { Component } from "react";

var CLASS_NAME = "coin";
var SELECTED_CLASS_NAME = "coin selected";

class Coin extends Component {

  constructor(props) {
    super(props);
    this.reunions = props.reunions;
  }

  selectMemberReunionsByType(e) {
    // Unselect all reunions on the map.
    this.resetMap();
    // Send the selected element to the parent App.
    this.props.setSelection(this);
    // Tag this coin as selected.
    this.setState({selected: true});
    var self = this;
    e.reunions.forEach(function(reunion) {
      if (reunion.category === self.props.type) {
        var marker = document.getElementById(reunion.element_id);
        marker.classList.add("picked");
      }
    })
  }

  resetMap() {
    var selectedReunions = map.querySelectorAll(".picked");
    selectedReunions.forEach(function(r) {
      r.classList.remove("picked");
    })
  }

  displayAsSelected() {
    if (this.props.selection && this.props.id === this.props.selection.props.id) {
      return true;
    }
    return false;
  }

  render() {
    return (
      <div className={this.displayAsSelected() ? SELECTED_CLASS_NAME : CLASS_NAME} onClick={()=>{this.selectMemberReunionsByType(this.props)}}>{this.props.counts}</div>
    )
  }
}

export default Coin;
