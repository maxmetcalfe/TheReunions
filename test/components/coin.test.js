import React from "react"
import renderer from "react-test-renderer"
import Coin from "../../src/components/Coin"

test("Coin", () => {
  let component = renderer.create(<Coin type="four"></Coin>)

  expect(component).toMatchSnapshot()
});