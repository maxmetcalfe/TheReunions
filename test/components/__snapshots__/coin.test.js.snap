// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin 1`] = `
<div
  className="coin four"
  onClick={[Function]}
/>
`;
